File: Deniz/day_1.rs
impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut stack: Vec<char> = Vec::new();
        let opening = ['(', '[', '{'];
        let closing = [')', ']', '}'];

        for c in s.chars() {
            if opening.contains(&c) {
                stack.push(c);
            } else {
                match stack.pop() {
                    Some(top) => {
                        let index_open = opening.iter().position(|&x| x == top);
                        let index_close = closing.iter().position(|&x| x == c);
                        if index_open != index_close {
                            return false;
                        }
                    },
                    None => return false,
                }
            }
        }

        stack.is_empty() 
    }
}

File: Deniz/day_1.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = list()
        opening = ['(', '[' ,'{']
        closing = [')', ']' ,'}']
        for i in s:
            if i in opening:
                stack.append(i)
            else:
                if len(stack) == 0:
                    return False 
                elif closing.index(i) == opening.index(stack[-1]):
                    stack.pop()
                else:
                    return False 

        return len(stack) == 0
